### PR TITLE
SVGLoader: Implement 'defs' and 'use' nodes

### LINKED ISSUE #14569: SVGLoader: Missing SVG specs break correct shape rendering.
I mean, I understand SVGLoader is work in progress, but can you really get anywhere without this? my svg looks exploded :)

example:
```xml
<g transform="translate(226 226)">
      <polygon fill="#F4F7FF" points="...
```

in SVGLoader there is only this:
```javascript
case 'g':
	style = parseStyle( node, style );
	break;
```

while funcs like `parseTransformNode` are clearly there already

### PR DESCRIPTION
**Pull request by yomboprime**
The ```defs``` node contains definitions to use later by the ```use``` node. It allows to use SVG files which contains svg clone objects.

Thanks to @TakumaKira for the expression to get the referenced node.

Fixes #14569

### CODE REVIEW

**Comment by yomboprime** [comment]
I've commited the jsm instead of the js, please wait while I fix this.

**Comment by yomboprime** [comment]
Done.

File: examples/js/loaders/SVGLoader.js [hunk 1]
```diff
@@ -641,13 +665,13 @@ THREE.SVGLoader.prototype = Object.assign( Object.create( THREE.Loader.prototype
 			rx = Math.abs( rx );
 			ry = Math.abs( ry );
 
-			// Compute (x1′, y1′)
+			// Compute (x1â², y1â²)
```
Comment (Mugen87) [inline examples/js/loaders/SVGLoader.js:668]: Um, this symbols look strange? 🤔 

BTW: Please run `node utils/modularize.js` once in order to update the jsm module.

Comment (yomboprime) [reply examples/js/loaders/SVGLoader.js:668]: I think they were accent chars before. I will change them to apostrophes. Must have been my editor, I saw a warning in git about characters.

Comment (yomboprime) [reply examples/js/loaders/SVGLoader.js:668]: > BTW: Please run `node utils/modularize.js` once in order to update the jsm module.

Should I commit the jsm? I've seen in the contribution guidelines that I shouldn't.

Comment (Mugen87) [reply examples/js/loaders/SVGLoader.js:668]: >  I've seen in the contribution guidelines that I shouldn't.

Maybe there is a misunderstanding. You should always commit both files. However the workflow is editing the `js` file and generate the `jsm`.

Comment (yomboprime) [reply examples/js/loaders/SVGLoader.js]: Understood. I've commited it.

**Comment by Mugen87** [comment]
It seems the test file `Defs4` produce this warning:
```js

**Comment by yomboprime** [comment]
> It seems the test file `Defs4` produce this warning:
> 
> ```
> three.module.js:8077 THREE.Color: Unknown color url(#3-a)
> ```
> 
> Is this relevant for this PR?

The ```fill=url(nodeId)``` seems another form of using other node. But as the used node is a gradient (not implemented), it doesn't affect here.

**Comment by yomboprime** [comment]
I will look into implementing the other form of referencing the node. It seems easy. The SVG spec says it is possible to reference external SVG files with ```url( svgFileUrl#nodeId )```. I would implement access to local nodes, like in the example SVG (no url, just the node id).

**Comment by yomboprime** [comment]
After looking more at it I've seen the ```url()``` form is mainly used to apply gradients to shapes, so I think it wouldn't be of use to implement it.

**Comment by yomboprime** [comment]
I've added a console warning when ```url()``` is encountered. Should I just remove that token in the SVG sample file?

**Comment by Mugen87** [comment]
> Should I just remove that token in the SVG sample file?

I think it's okay to keep it :+1:.

**Review by Mugen87 — APPROVED** [review]

File: examples/js/loaders/SVGLoader.js [hunk 2]
```diff
@@ -674,11 +698,11 @@ THREE.SVGLoader.prototype = Object.assign( Object.create( THREE.Loader.prototype
 			var cxp = q * rx * y1p / ry;
 			var cyp = - q * ry * x1p / rx;
 
-			// Step 3: Compute (cx, cy) from (cx′, cy′)
+			// Step 3: Compute (cx, cy) from (cxâ², cyâ²)
```
Comment (mrdoob) [inline examples/js/loaders/SVGLoader.js:701]: 🤔

File: examples/js/loaders/SVGLoader.js [hunk 3]
```diff
@@ -674,11 +698,11 @@ THREE.SVGLoader.prototype = Object.assign( Object.create( THREE.Loader.prototype
...
 			var cx = Math.cos( x_axis_rotation ) * cxp - Math.sin( x_axis_rotation ) * cyp + ( start.x + end.x ) / 2;
 			var cy = Math.sin( x_axis_rotation ) * cxp + Math.cos( x_axis_rotation ) * cyp + ( start.y + end.y ) / 2;
 
-			// Step 4: Compute θ1 and Δθ
+			// Step 4: Compute Î¸1 and ÎÎ¸
```
Comment (mrdoob) [inline examples/js/loaders/SVGLoader.js:705]: 🤔

Comment (yomboprime) [reply examples/js/loaders/SVGLoader.js:705]: What codification should have the source .js files? UTF-8 or other?

Comment (mrdoob) [reply examples/js/loaders/SVGLoader.js]: Should be UTF-8. Maybe `modularize.js` doesn't currently support that?

Comment (yomboprime) [reply examples/js/loaders/SVGLoader.js]: > Maybe `modularize.js` doesn't currently support that?

But the problem is in the .js, not the jsm.

Comment (yomboprime) [reply examples/js/loaders/SVGLoader.js]: Just tested, and modularize seems to behave correctly.

File: examples/js/loaders/SVGLoader.js [hunk 4]
```diff
@@ -1092,144 +1118,158 @@ THREE.SVGLoader.prototype = Object.assign( Object.create( THREE.Loader.prototype
 
 		function parseNodeTransform( node ) {
 
-			var transform = new THREE.Matrix3();
+			var transform = new Matrix3();
```
Comment (mrdoob) [inline examples/js/loaders/SVGLoader.js:1121]: Does this work?

File: examples/jsm/loaders/SVGLoader.js [hunk 5]
```diff
@@ -684,11 +708,11 @@ SVGLoader.prototype = Object.assign( Object.create( Loader.prototype ), {
 			var cxp = q * rx * y1p / ry;
 			var cyp = - q * ry * x1p / rx;
 
-			// Step 3: Compute (cx, cy) from (cx′, cy′)
+			// Step 3: Compute (cx, cy) from (cxâ², cyâ²)
```
Comment (mrdoob) [inline examples/jsm/loaders/SVGLoader.js:711]: Still seeing encoding issues...

File: examples/js/loaders/SVGLoader.js (hunk 4 above)
Comment (yomboprime) [reply examples/js/loaders/SVGLoader.js:1121]: My mistake. Fixed.

File: examples/jsm/loaders/SVGLoader.js (hunk 5 above)
Comment (yomboprime) [reply examples/jsm/loaders/SVGLoader.js]: Fixed also.

File: examples/js/loaders/SVGLoader.js (hunk 4 above)
Comment (mrdoob) [reply examples/js/loaders/SVGLoader.js:1121]: Hmm, are you sure it's fixed?

Comment (yomboprime) [reply examples/js/loaders/SVGLoader.js]: Sorry, I must have failed to git add it.

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**